### DISCUSSION TITLE
Enforce fixed column height

### DISCUSSION THREAD

**Discussion by e-tho** [YaLTeR/niri#573] [Ideas]
In a tiler like niri, it seems counter-intuitive for windows to overflow the screen or for a single window to occupy less than 100% of the screen height. Currently, there is no support for vertical scrolling, and windows are not in float mode, making overflow constraining. It makes more sense for the columns to be fully occupied, with other windows adjusting their size when one is increased in height.

**Comment by YaLTeR** [comment]
I agree that the current behavior sometimes feels bad. At some point I'll make a discussion listing design constraints and possible changes to the height handling, so we can find the best solution.

**Comment by e-tho** [comment]
@YaLTeR,
I noticed that when the height of a single window is adjusted, the windows are correctly resized. However, an overflow happens if the height of another window in the same column is altered. 
It's the modified state of one window that seems to break the resizing of the others, even reset to a height of 50%, from the moment it was modified this behavior occurs.
This issue seems more like a bug in the height management system rather than a design flaw.

**Reply by YaLTeR** [reply]
There's no bug. Windows can have either automatic height, or fixed (in pixels). When you resize a window it becomes fixed in height.

My reasoning was that when you change a window height, you set it to a particular height where something on the window is visible, so it must become fixed in pixels (rather than proportion, etc.).

**Reply by e-tho** [reply]
>My reasoning was that when you change a window height, you set it to a particular height where something on the window is visible, so it must become fixed in pixels (rather than proportion, etc.).

I feel like this behavior is well-suited to floating mode but doesn't quite fit with tiling mode.

**Reply by phisch** [reply]
Default for columns should be that windows add up to 100%, and there should be an exception for when you explicitly want to change that behavior.

I believe most cases expect this to add up to 100%. Every time I resize a window to be smaller or larger than 100%, It was on accident, and I have to reset the size. Same when I have multiple windows in a column, do a resize and close one window.

The current implementation is a hassle, and happens quite often. I'd rather have the case in which you need custom sizes to be a hassle, like explicitly unlocking a windows size and then being able to resize it.

**Comment by YaLTeR** [comment]
I made a new discussion where I wrote everything down, please suggest things there: https://github.com/YaLTeR/niri/discussions/593
